lib/omniauth/strategies/apple.rb

```diff
@@ -114,6 +114,13 @@ def last_name
         user_info.dig('name', 'lastName')
       end
 
+      def prune!(hash)
+        hash.delete_if do |_, v|
+          prune!(v) if v.is_a?(Hash)
+          v.nil? || (v.respond_to?(:empty?) && v.empty?)
+        end
+      end
```

Comment:
Thank you for introducing a good practice.
https://github.com/zquestz/omniauth-google-oauth2/blob/b23dc47975d8e8e6a1727a72ab22f096477d7688/lib/omniauth/strategies/google_oauth2.rb#L150-L155